pkg/transport/unix_listener.go

```diff
@@ -22,7 +22,7 @@ import (
 type unixListener struct{ net.Listener }
 
 func NewUnixListener(addr string) (net.Listener, error) {
-	if err := os.RemoveAll(addr); err != nil {
+	if err := os.Remove(addr); err != nil && !os.IsNotExist(err) {
```

Comment:
No. `os.IsNotExist(err)` is true if and only if the file to delete doesn't exist. `!os.IsExist(err)` would be true if and only if the operation failed for a reason other than the file existing (e.g., it would be true if the file did not exist). They're not interchangeable.